src/test/java/uk/gov/pay/directdebit/payments/services/sandbox/SandboxPaymentStateCalculatorTest.java

```diff
@@ -16,54 +23,123 @@
 import uk.gov.pay.directdebit.payments.model.SandboxPaymentId;
 import uk.gov.pay.directdebit.webhook.sandbox.resources.WebhookSandboxResource.SandboxEventAction;
 
+import java.time.ZonedDateTime;
 import java.util.Optional;
 
+import static java.time.ZoneOffset.UTC;
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
 import static org.mockito.BDDMockito.given;
+import static uk.gov.pay.directdebit.events.model.GovUkPayEventType.PAYMENT_SUBMITTED;
 import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.SANDBOX;
 import static uk.gov.pay.directdebit.mandate.fixtures.MandateFixture.aMandateFixture;
 import static uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture.aGatewayAccountFixture;
+import static uk.gov.pay.directdebit.payments.fixtures.GovUkPayEventFixture.aGovUkPayEventFixture;
 import static uk.gov.pay.directdebit.payments.fixtures.PaymentFixture.aPaymentFixture;
-import static uk.gov.pay.directdebit.payments.services.sandbox.SandboxPaymentStateCalculator.SANDBOX_ACTIONS_THAT_CHANGE_STATE;
+import static uk.gov.pay.directdebit.payments.services.GovUkPayEventToPaymentStateMapper.GOV_UK_PAY_EVENT_TYPES_THAT_CHANGE_PAYMENT_STATE;
+import static uk.gov.pay.directdebit.payments.services.sandbox.SandboxEventToPaymentStateMapper.SANDBOX_ACTIONS_THAT_CHANGE_PAYMENT_STATE;
 
-@RunWith(MockitoJUnitRunner.class)
+@RunWith(JUnitParamsRunner.class)
 public class SandboxPaymentStateCalculatorTest {
 
     private static final SandboxPaymentId SANDBOX_PAYMENT_ID = SandboxPaymentId.valueOf("Sandbox payment ID");
 
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
```

Comment:
This doesn't appear to be used anywhere, what is it for?

Reply:
It's needed because the class has parameterised tests so uses the parameterised runner rather than the Mockito runner. JUnit5 does this better I think because it gets rid of the idea of runners. We should use JUnit5!

Reply:
Ah I see, thanks for the explanation